discovery_test.go

```diff
@@ -0,0 +1,131 @@
+package oidc
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+type roundTripperFunc func(req *http.Request) (*http.Response, error)
+
+func (f roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+func Test_discover(t *testing.T) {
+	type args struct {
+		ctx          context.Context
+		issuer       string
+		httpClient   *http.Client
+		wellKnownUrl []string
+	}
+	tests := []struct {
+		name               string
+		args               args
+		want               *DiscoveryConfiguration
+		wantErr            bool
+		expectedBody       string
```

Comment:
It feels more natural to maybe call expectedBody and expectedStatusCode something like mockedResponseBody and mockedStatusCode or did I misinterpret how this works?

Reply:
Great point, the naming here is inconsistent. I am mixing expected/actual (common in Java and Go) with got/want (common in Go)

That being said, the body and the status code are neither mocks nor expected values I am comparing with but rather, perhaps, test fixtures or simply test data. I will see if I can come up with something more consistent for this.